lti/services/auth_data.py

```diff
@@ -36,60 +39,63 @@ class AuthDataService(object):
 
     """
 
-    def __init__(self, open_=None):
-        self.open_ = open_ or open  # Test seam.
-        self._name = 'canvas-auth.json'
-        self.auth_data = {}
-        self.load()
+    def __init__(self, db):
+        self._db = db
```

Comment:
We set `self._db` once, when the `AuthDataService` object is created, so that we don't have to pass `request.db` to `AuthDataService` methods each time we call them.